mystring.h

```diff
@@ -1,7 +1,19 @@
 #include <iostream>
 #include <memory>
 
-class String {
+class StringInterface {  
```

Comment:
I suggest putting the interface into a separate file

Reply:
done